code/game/turfs/simulated/floor/plating.dm

```diff
@@ -70,6 +70,42 @@
 			to_chat(user, "<span class='warning'>This section is too damaged to support a tile! Use a welder to fix the damage.</span>")
 		return TRUE
 
+	else if(istype(C, /obj/item/stack/sheet/glass))
+		if(broken || burnt)
+			to_chat(user, "<span class='warning'>Repair the plating first!</span>")
+			return TRUE
+		var/obj/item/stack/sheet/glass/R = C
+		if(R.get_amount() < 2)
+			to_chat(user, "<span class='warning'>You need two glass to make a glass floor!</span>")
```

Comment:
Two *sheets* of glass.

Reply:
oh god the grammar reviewer is bested at their own game